File: proyectos_c#/juegos/JuegoSnake/JuegoSnake/Objeto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JuegoSnake
{
    public class Objeto
    {
        public int x, y;
        public int tamano;
        public Objeto(int tamano, int x, int y)
        {
            this.x = x;
            this.y = y;
            this.tamano = tamano;
        }
        public Objeto(int tamano)
        {
            this.x = 0;
            this.y = 0;
            this.tamano = tamano;
        }
        public Boolean interseccion(Objeto obj)
        {
            var difx = Math.Abs(this.x - obj.x);
            var dify = Math.Abs(this.y - obj.y);
            if (difx >= 0 && difx < tamano && dify >= 0 && dify < tamano)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: proyectos_basic/juegos/SnakeBasic/SnakeBasic/Objeto.vb
Public Class Objeto
    Public x, y As Integer
    Public tamano As Integer
    Public Sub New(tamano As Integer, x As Integer, y As Integer)
        Me.x = x
        Me.y = y
        Me.tamano = tamano
    End Sub
    Public Sub New(tamano As Integer)
        Me.x = 0
        Me.y = 0
        Me.tamano = tamano
    End Sub
    Function interseccion(obj As Objeto)
        Dim difx As Integer
        difx = Math.Abs(x - obj.x)
        Dim dify As Integer
        dify = Math.Abs(y - obj.y)
        If (difx >= 0 And difx < tamano And dify >= 0 And dify < tamano) Then
            Return True
        Else
            Return False
        End If
    End Function
End Class
